packages/core/strapi/lib/types/factories.d.ts

```diff
@@ -28,11 +40,11 @@ type CollectionTypeRouterConfig = {
   delete?: HandlerConfig;
 };
 
-type RouterConfig = {
+type RouterConfig<T> = {
   prefix?: string;
-  only?: string[];
-  except?: string[];
-  config: SingleTypeRouterConfig | CollectionTypeRouterConfig;
+  only?: Array<keyof T>;
```

Comment:
My only regret here is that the given config won't be based on the related controller (giving access to custom methods).
What happens if we have a custom `helloWorld(): void` handler in the controller? Can we still use its name in the only or except ?

Reply:
Good point, do you suggest that we extend it to allow others or that we simply remove this type and leave it as a `string[]`?

Reply:
Tell me if it's Ok now or if you would remove the generic for the config too